### PR TITLE
Add sameSiteCookies option to SharedWorker options

### PR DESCRIPTION
**Pull request by arichiv**
### Description

This is the second of 3 PRs adding spec changes from https://privacycg.github.io/saa-non-cookie-storage/ launching in Chrome 125.

### Motivation

sameSiteCookies is a new option for SharedWorker construction that allows a given worker to be restricted to `SameSite=None` cookies even when created in a first-party context. This is important as only workers without access to `SameSite=Lax` and `SameSite=Strict` cookies can be accessed via requestStorageAccess.

### Additional details

https://chromestatus.com/feature/5175585823522816

### Related issues and pull requests

Relates to https://github.com/mdn/mdn/issues/543

### CODE REVIEW

**Comment by arichiv** [comment]
@sideshowbarker have time to take a look?

**Review by sideshowbarker — APPROVED** [review]

**Comment by arichiv** [comment]
@sideshowbarker I noticed the new spec I added shows up as 'unknown' and the older spec that was listed (for SharedWorker) is gone now. Is there a way I could restore the old spec and list the new one alongside it with a name?

File: files/en-us/web/api/sharedworker/sharedworker/index.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ short-title: SharedWorker()
 slug: Web/API/SharedWorker/SharedWorker
 page-type: web-api-constructor
 browser-compat: api.SharedWorker.SharedWorker
+spec-urls:
+  - https://html.spec.whatwg.org/multipage/workers.html#dom-sharedworker-dev
+  - https://privacycg.github.io/saa-non-cookie-storage/
```
Comment (sideshowbarker) [inline files/en-us/web/api/sharedworker/sharedworker/index.md:9]:
```suggestion
```

**Comment by arichiv** [comment]
I'm okay with this as is, how do I follow up to add compat information for sameSiteCookies? It's Chrome 125+ only right now

**Comment by sideshowbarker** [comment]
> @sideshowbarker I noticed the new spec I added shows up as 'unknown' and the older [spec](https://html.spec.whatwg.org/multipage/workers.html#dom-sharedworker-dev) that was listed (for SharedWorker) is gone now. Is there a way I could restore the old (default) spec and list the new one alongside it with a name?

Yes, the way to do that is to edit the file at https://github.com/mdn/browser-compat-data/blob/3bfaadfe0e5f56016ef782c5247f0cfc5a4dfada/api/SharedWorker.json#L67 and change the `spec_url` value there to an array that includes/adds any additional spec URLs.

**Comment by sideshowbarker** [comment]
> I'm okay with this as is, how do I follow up to add compat information for sameSiteCookies? It's Chrome 125+ only right now

That’d need to go somewhere in the data at https://github.com/mdn/browser-compat-data/blob/3bfaadfe0e5f56016ef782c5247f0cfc5a4dfada/api/SharedWorker.json#L63

**Comment by sideshowbarker** [comment]
Ari, let’s go ahead and land this. Any further changes can be handled as follow-up PRs here or in the browser-compat-data repo.

Congrats on landing your first docs change here — welcome aboard 🎉

**Pull request merged**